web-server/sgx-wallet-impl/src/schema/actions.rs

```diff
@@ -188,6 +194,8 @@ impl From<UnlockWalletError> for SaveOnfidoCheckResult {
         match err {
             InvalidWalletId => Self::InvalidAuth,
             InvalidAuthPin => Self::InvalidAuth,
+            //FIXME: Do additional error handling to prevent panic.
+            InvalidOtp => unreachable!(), // This variant should never happen as otp is not a feature required for KYC.
```

Comment:
I don't think this type of error is needed here as OTP errors will only occur in Payment transactions.

Reply:
When I don't include Invalidotp in the `match err`, I get the following
error: **_error[E0004]:_ non-exhaustive patterns: InvalidOtp not covered_**

In my latest commit, I've added a todo()! macro for now.

Reply:
You get the error because the enum `UnlockWalletError` has a arm for `InvalidOtp` - https://github.com/ntls-io/nautilus-wallet/blob/9529dd471a97995966de42f69c083733ed9664c2/web-server/sgx-wallet-impl/src/wallet_operations/store.rs#L97.

Think we can just remove that.

Reply:
Actually ignore that, I see that you bundled the OTP functionality with unlocking the wallet - https://github.com/ntls-io/nautilus-wallet/blob/9529dd471a97995966de42f69c083733ed9664c2/web-server/sgx-wallet-impl/src/wallet_operations/store.rs#L64. Therefore you have to have the `InvalidOtp` enum there.

I don't think we should combine unlocking the wallet with the OTP, as the OTP is just used for deciding whether to sign the transaction or not.

Reply:
Should I rather create a `UnlockWalletErrorWithOtp` enum perhaps?
As the `UnlockWalletError` enum is needed when OTP is checked by the user's response.

Reply:
You can or you can just leave as it is for now.

I thought that comparing the OTP wouldn't require the method to  "unlock wallet".

Reply:
I think we should leave it for now.

To compare OTP, would we not need to first unlock the wallet first, then compare OTP that's stored in the wallet against the one provided.

I tried implementing tests but I can't finish it in time. I plan to write `generate_otp` and `sign_transaction_with_otp` tests at a later time.

Reply:
To access the OTP, the enclave doesn't need the pin to access it - https://github.com/ntls-io/nautilus-wallet/blob/9529dd471a97995966de42f69c083733ed9664c2/web-server/sgx-wallet-impl/src/wallet_operations/store.rs#L66.

But let's see how this way works first.

The tests are important to check/ensure that they work as intended. If we skip the tests, we only check if these backend methods work when we integrate it with the frontend.